wazuh/wazuh-splunk
# CDB list input validation

Summary
---------
Closes #1252 

This PR adds input data validation for the CDB lists.

- Validates the CDB list name by using this regular expression: `^[\\w\\.\\-]+$`
- Validates the CDB key:value pairs
  - Keys: `?:^"([\\w\\-:]+?)"|^[^:"\\s]+$)`
  - Values: `(?:^"([\\w\\-:]*?)"$|^[^:"]*$)`

This PR also renames the CDB List navigation item from 'Lists' to 'CDB Lists' to make it more descriptive and consistent with the Kibana App.

To test
-------

- Check that the navigation item to go to the CDB lists is named as 'CDB Lists' across all subsections of the Ruleset section.
- Check that CDB list with invalid name cannot be created.
    - Can only contain alphanumeric characters, dots (.), hyphens (-) and underscores (_).
- Check that no key:pair value can be created using an invalid **key** (invalid values are replaced with an empty string).
  - Examples [here](https://github.com/wazuh/wazuh/blob/eaab105fb5a7430173a52a2598deef204b06e809/framework/wazuh/core/cdb_list.py#L185).
